lib/plugins/leetcode.js

```diff
@@ -552,11 +552,36 @@ function parseCookie(cookie, cb) {
   };
 }
 
-function saveAndGetUser(user, cb, cookieData) {
-  user.sessionId = cookieData.sessionId;
-  user.sessionCSRF = cookieData.sessionCSRF;
-  session.saveUser(user);
-  plugin.getUser(user, cb);
+// leetcode-cn.com Cookie is not the same as leetcode.com in third parties
+function parseCNCookie(cookie, body, cb) {
+  const SessionPattern = /LEETCODE_SESSION=(.+?)(;|$)/;
+  const csrfPattern = /name="csrfmiddlewaretoken" value="(.*?)"/;
+  const reSessionResult = SessionPattern.exec(cookie);
+  const reCsrfResult = csrfPattern.exec(body);
+  if (reSessionResult === null || reCsrfResult === null) {
+    return cb('invalid cookie?');
+  }
+  return {
+    sessionId:   reSessionResult[1],
+    sessionCSRF: reCsrfResult[1],
+  };
+}
+
+function requestLeetcodeAndSave(request, leetcodeUrl, user, cb, party) {
+  request.get({url: leetcodeUrl}, function(e, resp, body) {
+    const redirectUri = resp.request.uri.href;
+    if (redirectUri !== config.sys.urls.leetcode_redirect) {
+      return cb(`${party} login failed or ${party} did not connect to LeetCode`);
```

Comment:
Is it necessary to pass a parameter only for logging here? How about just change it to:

`Login failed. Please make sure the credential is correct.`

I'm thinking that we can give more detailed hint from the VS Code extension side.

Reply:
Of course~